server/src/services/restaurant.ts

```diff
@@ -34,11 +34,17 @@ async function updateMenu(
 }
 
 type Meters = number;
-async function findNear(
-  coordinates: Coordinates,
-  searchRadius: Meters
-): Promise<RestaurantDocument[] | null> {
-  return null;
+async function findNear(coordinates: Coordinates,searchRadius: Meters): Promise<RestaurantDocument[] | null> {
+  return RestaurantModel.aggregate([
+    {
+      $geoNear:{
+        near: { type: "Point", coordinates: coordinates },
+        distanceField: "dist.calculated",
```

Comment:
Let's call this field "distance" instead. 
With "dist.calculated", a resulting restaurant would look like:
```ts
[{
  _id: ObjectId(...),
  yelp_id: "...",
  ...,
  dist : { calculated: 3000 }
}, ...]
```

if it's just "distance", then it would look like this:
```ts
[{
  _id: ObjectId(...),
  yelp_id: "...",
  ...,
  distance : 3000
}, ...]
```